File: ssd_app/my_custom/report/received_details_update/received_details_update.py
# Copyright (c) 2025, SSDolui and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
    filters = filters or {}
    status_filter = filters.get("status", "Pending")  # default Pending

    # Build WHERE clause based on status
    where_clause = ""
    if status_filter == "Pending":
        where_clause = "WHERE rec.rec_details != 1"
    elif status_filter == "Updated":
        where_clause = "WHERE rec.rec_details = 1"
    # "All" -> no WHERE clause

    query = f"""
       SELECT 
            rec.name,
            cif.inv_no, 
            rec.received_date,
            cif.inv_date, 
            cus.customer, 
            noti.notify, 
            bank.bank,
            IF(cif.payment_term IN ('LC', 'DA'),
                CONCAT(cif.payment_term, '- ', cif.term_days),
                cif.payment_term
            ) AS p_term,
            cif.document,
            rec.received,
            rec.rec_details
        FROM `tabDoc Received` rec
        LEFT JOIN `tabCIF Sheet` cif ON cif.name = rec.inv_no
        LEFT JOIN `tabCustomer` cus ON cus.name = rec.customer
        LEFT JOIN `tabNotify` noti ON noti.name = rec.notify
        LEFT JOIN `tabBank` bank ON bank.name = rec.bank
		{where_clause}
		ORDER BY rec.received_date DESC
    """

    data = frappe.db.sql(query, as_dict=1)

    columns = [
        {"label": "Inv No", "fieldname": "inv_no", "fieldtype": "Data", "width": 125},
        {"label": "Inv Date", "fieldname": "inv_date", "fieldtype": "Date", "width": 110},
        {"label": "Rec Date", "fieldname": "received_date", "fieldtype": "Date", "width": 110},
        {"label": "Customer", "fieldname": "customer", "fieldtype": "Data", "width": 250},
        {"label": "Notify", "fieldname": "notify", "fieldtype": "Data", "width": 250},
        {"label": "Bank", "fieldname": "bank", "fieldtype": "Data", "width": 70},
        {"label": "P Term", "fieldname": "p_term", "fieldtype": "Data", "width": 95},
        {"label": "Document", "fieldname": "document", "fieldtype": "Float", "width": 125},
        {"label": "Rec Amnt", "fieldname": "received", "fieldtype": "Float", "width": 125},
    ]

    return columns, data

File: ssd_app/my_custom/report/received_details_update/received_details_update.js
// Copyright (c) 2025, SSDolui and contributors
// For license information, please see license.txt

frappe.query_reports["Received Details Update"] = {
	 "filters": [
        {
            "fieldname": "status",
            "label": __("Status"),
            "fieldtype": "Select",
            "options": ["All", "Updated", "Pending"],
            "default": "Pending",
            "reqd": 1
        }
    ],
	formatter: function (value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);

        if (column.fieldname === "inv_no" && data && data.inv_no && data.nego_details != 1) {
            const doctype = "Doc Received Details";
            const inv_no = String(data.name).replace(/'/g, "\\'");

            return `
                ${value}
                <a href="#"
                   title="Create ${doctype} for ${inv_no}"
                   style="margin-left:6px; color:#007bff; text-decoration:none;"
                   onclick="frappe.route_options = { inv_no: '${inv_no}' }; frappe.set_route('Form', '${doctype}', 'new-${doctype}'); return false;">
                   <i class="fa fa-plus-circle"></i>
                </a>`;
        }

        return value;
    }
};
